graphql/exec.go

```diff
@@ -21,12 +21,75 @@ func CollectFields(doc *query.Document, selSet []query.Selection, satisfies []st
 	return collectFields(doc, selSet, satisfies, variables, map[string]bool{})
 }
 
+func runDirectives(directives common.DirectiveList, variables map[string]interface{}) bool {
+	for _, directive := range directives {
+		if !runSingleDirective(directive, variables) {
+			return false
+		}
+	}
+	return true
+}
+
+func runSingleDirective(directive *common.Directive, variables map[string]interface{}) bool {
+	arg := directive.Args.MustGet("if")
+	fieldAllowed := true
+	switch arg.(type) {
+	case *common.Variable:
+		{
+			value := arg.Value(variables)
+			if realValue, ok := value.(bool); ok {
+				fieldAllowed = realValue
+			}
+		}
+	case *common.BasicLit:
+		{
+			value := arg.Value(nil)
+			if realValue, ok := value.(bool); ok {
+				fieldAllowed = realValue
+			}
+		}
+	}
+
+	switch directive.Name.Name {
+	case "skip":
+		{
+			return !fieldAllowed
+		}
+	case "include":
+		{
+			return fieldAllowed
+		}
+	default:
+		// Custom directives are not supported yet
+		return fieldAllowed
```

Comment:
probably should `return true`. Do nothing rather then do something unexpected. Or error.